crates/llvm-context/src/polkavm/mod.rs

```diff
@@ -22,64 +25,51 @@ pub fn initialize_target() {
 /// Builds PolkaVM assembly text.
 pub fn build_assembly_text(
     contract_path: &str,
-    assembly_text: &str,
+    encoded_hex_text: &str,
     _metadata_hash: Option<[u8; revive_common::BYTE_LENGTH_WORD]>,
     debug_config: Option<&DebugConfig>,
 ) -> anyhow::Result<Build> {
     if let Some(debug_config) = debug_config {
-        debug_config.dump_assembly(contract_path, assembly_text)?;
+        debug_config.dump_assembly(contract_path, encoded_hex_text)?;
     }
 
-    /*
-    let mut assembly =
-        zkevm_assembly::Assembly::from_string(assembly_text.to_owned(), metadata_hash).map_err(
-            |error| {
-                anyhow::anyhow!(
-                    "The contract `{}` assembly parsing error: {}",
-                    contract_path,
-                    error,
-                )
-            },
-        )?;
-
-    let bytecode_words = match zkevm_assembly::get_encoding_mode() {
-        zkevm_assembly::RunningVmEncodingMode::Production => { assembly.compile_to_bytecode_for_mode::<8, zkevm_opcode_defs::decoding::EncodingModeProduction>() },
-        zkevm_assembly::RunningVmEncodingMode::Testing => { assembly.compile_to_bytecode_for_mode::<16, zkevm_opcode_defs::decoding::EncodingModeTesting>() },
-    }
+    let bytecode = hex::decode(encoded_hex_text)
+        .map_err(|e| anyhow::anyhow!("Failed to decode encoded hex text:\n{}\n", e))?;
+
+    let program_blob = ProgramBlob::parse(bytecode.as_slice())
+        .map_err(|error| anyhow::anyhow!(format!("Failed to parse program blob:\n{}\n", error)))?;
+
+    let mut disassembler =
+        Disassembler::new(&program_blob, DisassemblyFormat::Guest).map_err(|error| {
+            anyhow::anyhow!(format!(
+                "Failed to create disassembler for contract:\n{:?}\n\nDue to:\n{}\n",
+                contract_path, error
+            ))
+        })?;
+
```

Comment:
```suggestion
    disassembler.display_gas()?;
```

We always want to display the gas usage, not only on error :)